### ISSUE TITLE
Sponge mod 

### ISSUE THREAD

**Issue by Klaranth** [pandorabox-io/in-game#92]
Ruggila :
Should we install the sponge mod?  
https://forum.minetest.net/viewtopic.php?t=20729
This will be helpful to solve problems too big for quick cleaning, too small for worldedit. 
(Example : An artificial sea placed at another player's home base.

**Comment by wsor4035** [comment]
do note that its air like nodes can sometimes cause issues

**Comment by Omikhleia** [comment]
Quickly checked this sponge mod in single-player. Not sure what it is supposed to address or solve, but...
- The fact the sponges work underwater (even real deep), creating a large illogical bubble of air near to immediately, kills suspension of disbelief for me...
- It also makes it very (far too) easy to build an underwater base. Just put an array of these sponges, et voilà!
- (As for the corals and the kelp so easily exposed, or the floating waterlilies: well of course this can already be achieved in usual gameplay, though a bit less easily maybe.). 

In other terms, I am not convinced at all it adds anything worth to the gameplay... 

(If at least they'd stop working at some point when soggy enough, so one has to fight water as we have to fight vacuum somewhere, well there would be a bit of challenge)

**Comment by S-S-X** [comment]
Agreed it sounds like some improvements for survival style game play would be very good for this.

> The fact the sponges work underwater (even real deep), creating a large illogical bubble of air near to immediately, kills suspension of disbelief for me...

Check some area above sponge and let it work only if there's no water above it?

> If at least they'd stop working at some point when soggy enough

For me this also sounds like very good addition.

I think it would also be good if it could work without those sponge:liquid_stop nodes and would not remove all water immediately but maybe few source nodes / every few server steps (nodetimer with 0.25 second timer should work fine).

Could fork https://github.com/BenjieFiftysix/sponge (MIT license) and try some simple modifications like that, I might try few things to see if my suggestions would work at all.

**Comment by Klaranth** [comment]
Yes - 8 
No  - 2

VoteBot@bot> Votes for "Install Sponge Mod":
 Klaranth — Yes
 PhrogMan — Yes
 SamWorld — Yes
 xman — Yes
 Lejo — No
 SimonSays — Yes
 Vamp — Yes
 marghl — Yes
 Eula — Yes
 TesterGuy — No

**Comment by Klaranth** [comment]
Sher Khan 
Add sponge to remove water
Because, let's say there is a water while making a base underground and there is a large water body in mining area, difficult to remove by a solo player 

SX 
Few things you can do: You can replace water with replacer, but if you do it is best to first test in small area how replacer works underwater.
Or you can move water away with jumpdrive. Or you can remove water with digtron.

Sher Khan 
I didn't know about the replacer (so no comment), but jp may not fit, or digtron traction error will occur

SX 
Digtron wont need any traction when going downwards...
For replacer, try replacing with some nodes that drop without supporting node below them...

BuckarooBanzai 
last time i checked the sponge mod had trouble with the vacuum or some other mod, not sure but there were some issues, feel free to test and report back 👍

**Comment by Klaranth** [comment]
Huhhila
 I'd like to have something similar to using replacer in crust mode for replacing liquids with tnt:gunpowder and then lighting it up. Preferably waterproof, abm-less and limited in range... lets say to half of what beacon beams can be. Dunno what kind of havoc water-reactive and pyrophoric materials in general would cause.
